src/main/java/bob/Parser.java

```diff
@@ -0,0 +1,99 @@
+package bob;
+
+/**
+ * Represents a parser that deals with making sense of the user command
+ */
+public class Parser {
+
+    public Parser() {
+
+    }
+
+    public void parse(String input) {
+
+    }
+
+    /**
+     * Checks if the input statement is a command to mark a Task
+     *
+     * @param input
+     * @return boolean true if input is mark
+     */
+    public boolean isMark(String input) {
+        char[] charArray = input.toCharArray();
+        if (charArray[0] == 'm' && charArray[1] == 'a' && charArray[2] == 'r' && charArray[3] == 'k'
+                && Character.isWhitespace(charArray[4]) && Character.isDigit(charArray[5])) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns the digit of the index of the Task that should be marked
+     *
+     * @param input
+     * @return int the index of marked task
+     */
+    public int getMarkDigit(String input) {
+        char[] charArray = input.toCharArray();
+        return Character.getNumericValue(charArray[5]);
+    }
+
+    /**
+     * Checks if input statement is a command to delete a Task
+     *
+     * @param input
+     * @return boolean true if statment is a delete command
+     */
+    public boolean isDelete(String input) {
+        char[] charArray = input.toCharArray();
+        if (charArray[0] == 'd' && charArray[1] == 'e' && charArray[2] == 'l' && charArray[3] == 'e' &&
```

Comment:
Perhaps there is a better way to do this than brute-force it here and in other places? You could look into regex (regular expression) or other methods of doing this easier, so as to avoid complicated expressions which is one of the guidelines for [code quality](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html).